cpp/examples/basic/src/process_csv.cpp

```diff
@@ -0,0 +1,69 @@
+#include <cudf/aggregation.hpp>
+#include <cudf/groupby.hpp>
+#include <cudf/io/csv.hpp>
+#include <cudf/table/table.hpp>
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+std::unique_ptr<cudf::table> read_csv(std::string const& file_path)
+{
+  auto source_info    = cudf::io::source_info(file_path);
+  auto builder        = cudf::io::csv_reader_options::builder(source_info);
+  auto options        = builder.build();
+  auto data_with_meta = cudf::io::read_csv(options);
+  return std::move(data_with_meta.tbl);
```

Comment:
I don't think the move is necessary here.  Can just collapse these two lines: 
```suggestion
  return cudf::io::read_csv(options);
```

Reply:
So `read_csv` returns a `table_with_metadata` struct, which is the owner of the table. We could instead return the `data_with_meta` (which I think that's pretty neat as well). Updated with that.

Reply:
Sorry, I read the `.tbl` as `_tbl` and missed that you were extracting a field of a struct.

Reply:
I think part of the problem is you use `auto` everywhere so the types are not obvious. That's fine, but in examples sometimes it's nice to see the types to aid learning the API.